filters/imagefilter.go

```diff
@@ -57,17 +57,18 @@ func errorResponse() *http.Response {
 }
 
 func HandleImageResponse(ctx filters.FilterContext, f ImageFilter) {
-	//executed while processing the first filter
-	if _, ok := ctx.StateBag()[SkropInit]; !ok {
-		initResponse(ctx)
-		ctx.StateBag()[SkropInit] = true
-	}
 
 	//in case the response had an error from the backend or from a previous filter
 	if ctx.Response().StatusCode > 300 {
```

Comment:
changed the order of the block to handle the status code properly, I don't think we have to process anything here if we get a status code other than 200.